docs/bundler.md

```diff
@@ -0,0 +1,112 @@
+# bundler
+
+<https://bundler.io/>
+
+## Prerequisites
+
+To use Cachi2 with Bundler locally, ensure you have Ruby and Bundler installed on your system.
+
+```bash
+sudo dnf install rubygem-bundler
+```
+
+Or use the official Ruby image for example from Docker Hub:
+
+```bash
+podman run --rm -it docker.io/library/ruby:latest bash
+```
+
+Then ensure you have both, **Gemfile** and **Gemfile.lock** in your project directory. Anything outside
+of the **Gemfile.lock** will be ignored. **Gemfile** is only necessary when parsing the **Gemfile.lock**
+to pre-fetch all dependencies specified in your project's **Gemfile.lock**. Please note, that we don't
+use any Bundler commands to pre-fetch dependencies. Instead, we use native mechanisms from the Ruby standard
+library to parse the **Gemfile.lock** and download the dependencies to the specified output directory.
+
+## Usage
+
+Run the following command in your terminal to pre-fetch your project's dependencies.
+The command will download all dependencies specified in the **Gemfile.lock** to the specified output directory.
+
+```bash
+cachi2 fetch-deps --source ./path-to-your-ruby-project --output ./cachi2-output bundler
+```
+
+### Configuration
+
+Bundler uses an unorthodox system when dealing with configuration options. The highest precedence
+is given to the config file, and then to the environment variables. See the official documentation:
+<https://bundler.io/v2.5/man/bundle-config.1.html#DESCRIPTION>>
+
+The order of precedence for Bundler configuration options is as follows:
+
+1. Local config (`<project_root>/.bundle/config or $BUNDLE_APP_CONFIG/config`)
+2. Environment variables (ENV)
+3. Global config (`~/.bundle/config`)
+4. Bundler default config
+
+Since the local configuration takes higher precedence than the environment variables
+(except `BUNDLE_APP_CONFIG`), we need to set a few Bundler configuration options to make the build
+work by copying the configuration file and setting the environment variables below. Then, we change
+the `BUNDLE_APP_CONFIG` environment variable to point to the copied/updated configuration file.
+
+This is an unfortunate limitation of Bundler, and we hope that Bundler will change this behavior in
```

Comment:
I would suggest dropping the first sentence of the paragraph.

Reply:
So we don't want to point out to Bundler as a source of the problem ? :disappointed:

Reply:
I would actually prefer if we kept
`This is an unfortunate limitation of Bundler` in, but dropped the rest of the sentence, because that would be a breaking change for Bundler. I would not expect a package ecosystem toolkit introducing such a breaking change. Besides, our use case for it is very niche to justify such a change with majority of direct Bundler users likely not being affected at all despite being a very weird and definitely uncommon pattern to see.

Reply:
On a side note, hoping for a change in Bundler without raising awareness by opening an issue will likely never end up being fulfilled, so if you wish to keep the whole sentence, we should back it with a reference to an upstream issue.

Reply:
Keeping the first part is a good idea, I would have dropped 'unfortunate' though: it simultaneously overly personalizes the text and hints that there could be fortunate limitations.